runtime/amd64.S

```diff
@@ -112,14 +112,14 @@
 
 #endif
 
-        .set    domain_curr_field, 0
+        .set    caml_domain_curr_field, 0
 #define DOMAIN_STATE(c_type, name) \
```

Comment:
The `domain_xxx` absolute symbols are local symbols: `nm` reports them as `a` (lowercase = local) rather than `A` (uppercase = global).  Hence, there is no need to prefix these symbols.  Instead, `check_symbols` must be fixed to ignore all local symbols.